applications/body_pose_estimation/CMakeLists.txt

```diff
@@ -12,34 +12,44 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+project(body_pose_estimation NONE)
 
 find_package(holoscan 1.0 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
 # Download the yolov8 pose model if it doesn't exist
-if(NOT EXISTS "${HOLOHUB_DATA_DIR}/body_pose_estimation/yolov8l-pose.onnx")
-    add_custom_target(yolov8_onnx ALL
-      COMMAND yolo export model=yolov8l-pose.pt format=onnx
-      COMMAND mkdir -p "${HOLOHUB_DATA_DIR}/body_pose_estimation"
-      COMMAND mv yolov8l-pose.onnx "${HOLOHUB_DATA_DIR}/body_pose_estimation"
-      BYPRODUCTS "yolov8l-pose.onnx"
+add_custom_command(OUTPUT "${HOLOHUB_DATA_DIR}/body_pose_estimation/yolov8l-pose.onnx"
+    COMMAND yolo export model=yolov8l-pose.pt format=onnx
+    COMMAND mkdir -p "${HOLOHUB_DATA_DIR}/body_pose_estimation"
+    COMMAND mv yolov8l-pose.onnx "${HOLOHUB_DATA_DIR}/body_pose_estimation"
+    BYPRODUCTS "${HOLOHUB_DATA_DIR}/body_pose_estimation/yolov8l-pose.onnx"
     )
-endif()
 
-# Convert video to gxf entities if needed
-if(NOT EXISTS "${HOLOHUB_DATA_DIR}/body_pose_estimation/twirl.gxf_entities" OR
-   NOT EXISTS "${HOLOHUB_DATA_DIR}/body_pose_estimation/twirl.gxf_index")
-    # Download video file from Pexels
-    execute_process(
-        COMMAND curl -L -o "${HOLOHUB_DATA_DIR}/body_pose_estimation/twirl.mp4"
-            "https://www.pexels.com/download/video/5385885/?fps=25.0&h=1920&w=1080"
+# Download the testing video
+add_custom_command(OUTPUT "${HOLOHUB_DATA_DIR}/body_pose_estimation/twirl.mp4"
+    COMMAND mkdir -p "${HOLOHUB_DATA_DIR}/body_pose_estimation"
+    COMMAND curl -S -o "${HOLOHUB_DATA_DIR}/body_pose_estimation/twirl.mp4"
+              -L "https://www.pexels.com/download/video/5385885/?fps=25.0&h=1920&w=1080"
+    BYPRODUCTS "${HOLOHUB_DATA_DIR}/body_pose_estimation/twirl.mp4"
+    VERBATIM
```

Comment:
@jjomier is the `VERBATIM` flag needed for correct behavior?

Reply:
Actually yes - otherwise the curl command runs in a separate thread (not sure why).